src/main/java/com/ryanharter/auto/value/parcel/AutoValueParcelExtension.java

```diff
@@ -95,93 +126,87 @@ String getParcelableCode(Context context, String className, String classToExtend
     return javaFile.toString();
   }
 
-  private void validateProperties(ProcessingEnvironment env,
-                                  Map<String, ExecutableElement> properties) {
+  private List<Property> readProperties(Map<String, ExecutableElement> properties) {
+    List<Property> values = new LinkedList<Property>();
+    for (Map.Entry<String, ExecutableElement> entry : properties.entrySet()) {
+      values.add(new Property(entry.getKey(), entry.getValue()));
+    }
+    return values;
+  }
+
+  private void validateProperties(ProcessingEnvironment env, List<Property> properties) {
     Types typeUtils = env.getTypeUtils();
-    Elements elementUtils = env.getElementUtils();
-    TypeMirror list = typeUtils.getDeclaredType(
-        elementUtils.getTypeElement(List.class.getName()),
-        typeUtils.getWildcardType(null, null)
-    );
-    TypeMirror parcelable = env.getElementUtils().getTypeElement("android.os.Parcelable").asType();
-    TypeMirror serializable = TypeSimplifier.typeFromClass(env.getTypeUtils(),
-        env.getElementUtils(), Serializable.class);
-    for (ExecutableElement name : properties.values()) {
-      TypeMirror type = name.getReturnType();
-      if (type.getKind() == TypeKind.DECLARED && typeUtils.isSubtype(type, list)) {
-        DeclaredType dType = (DeclaredType) type;
-        List<? extends TypeMirror> types = dType.getTypeArguments();
-        if (!types.isEmpty()) {
-          type = types.get(0);
-        }
-      } else if (type.getKind() == TypeKind.ARRAY) {
+    for (Property property : properties) {
+      TypeMirror type = property.element.getReturnType();
+      if (type.getKind() == TypeKind.ARRAY) {
         ArrayType aType = (ArrayType) type;
         type = aType.getComponentType();
       }
-
-      if (!TypeName.get(type).isPrimitive() &&
-          !TypeSimplifier.isClassOfType(env.getTypeUtils(), parcelable, type) &&
-          !TypeSimplifier.isClassOfType(env.getTypeUtils(), serializable, type)) {
+      TypeElement element = (TypeElement) typeUtils.asElement(type);
+      if ((element == null || !Parcelables.isValidType(typeUtils, element)) &&
+          !Parcelables.isValidType(TypeName.get(type))){
         env.getMessager().printMessage(Diagnostic.Kind.ERROR, "AutoValue property " +
-            name.getSimpleName() + " is not primitive, Parcelable, or Serializable.", name);
+            property.name + " is not a supported Parcelable type.", property.element);
         throw new AutoValueParcelException();
       }
     }
   }
 
+  private static final List<String> METHOD_EXCLUDES =
+      Arrays.asList("describeContents", "toBuilder");
```

Comment:
Actually because you're calling `.contains` it should be a `Set` (or `ImmutableSet`)